//  Copyright 2022 Google LLC
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//       http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.

syntax = "proto3";

package kv_server.tools.bidding_auction_data_generator;

import "google/protobuf/struct.proto";

//SideLoadData proto to convert side load data in Json to proto message
//to allow easy parsing of custom audience names and render_urls.
message SideLoadData {
  message CustomAudienceData {
    // Name of interest group/custom audience.
    string name = 1;
    // list of Ad
    repeated Ad ads = 2;
  }
  //list of interest groups
  repeated CustomAudienceData interest_groups = 1;
}

message Ad {
  //render_url for the buyer
  string render_url = 1;
  //meta data in random JSON array.
  google.protobuf.ListValue metadata = 2;
}

// This proto is the original BYOS GetValuesResponse proto.
// It is different from KV V1 GetValuesResponse proto in public/query folder
// The V1 proto is no longer compatible with original BYOS response,
// as it now returns the reason why the value is not returned.
message BYOSGetValuesResponse {
  // Map of key value pairs for namespace keys.
  google.protobuf.Struct keys = 1;

  // Map of key value pairs for namespace renderUrls.
  google.protobuf.Struct render_urls = 2;

  // Map of key value pairs for namespace adComponentRenderUrls.
  google.protobuf.Struct ad_component_render_urls = 3;
}
